api.py

```diff
@@ -69,70 +69,80 @@
 #Sign_up
 @app.route("/auth/sign_up", methods=["GET", "POST"])
 def sign_up():
-    result =  auth.sign_up( request.form["firstname"], request.form["lastname"], request.form["email"], request.form["password"], request.form["confirmpassword"] )
 
     if request.method == "POST":
-
-        return result
+        result =  auth.sign_up( request.form["firstname"], request.form["lastname"], request.form["email"], request.form["password"], request.form["confirmpassword"] )
+        
+        if "error" in result:
+            result =  auth.sign_up( request.form["firstname"], request.form["lastname"], request.form["email"], request.form["password"], request.form["confirmpassword"] )
+
+            serviceutils.log_event("sign_up_error", {
+                        "uid": result["uid"],
+                        "error": result["error"]
+                        })
+        else:
+            serviceutils.log_event("user_signed_up", {
+                        "uid": result["uid"]   
+                        })
     
-    if "error" in result:
-
-        serviceutils.log_event("sign_up_error", {
-                    "uid": result["uid"],
-                    "error": result["error"]
-                    })
-
-    serviceutils.log_event("user_signed_up", {
-                    "uid": result["uid"]   
-                    })
+            
+            return sign_up_html
```

Comment:
The "return sign_up_html" should be back where it was. It is not relevant if the request is a post request.

Same with all the other ones.